### PR TITLE
docs(core): add kubernetes setup guide

### PR DESCRIPTION
**Pull request by isaacplmann**
Adds a kubernetes set up guide for nx cloud on prem

### CODE REVIEW

**Comment by isaacplmann** [comment]
Once this is merged, we should update the readme in nx-cloud-helm to redirect people to this guide so that content only needs to be updated in one place

File: docs/nx-cloud/private/kubernetes-setup.md [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+  tag: 'latest'
+
+nxCloudAppURL: 'https://nx-cloud.myorg.com'
+
+ingress:
+  skip: true
+```
+
+<details>
+<summary>⤵️ and then define it yourself (expand me)</summary>
```
Comment (StalkAltan) [inline docs/nx-cloud/private/kubernetes-setup.md:131]: The <details> and <summary> tags aren't properly rendered on nx.dev. We can probably just remove them and show the full content.

File: docs/nx-cloud/private/kubernetes-setup.md [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+                             - path: /download
+          - pathType: Prefix
+            backend:
+               service:
+                  name: nx-cloud-frontend-service
+                  port:
+                     number: 8080
+```
+
+</details>
```
Comment (StalkAltan) [inline docs/nx-cloud/private/kubernetes-setup.md:198]: Can be removed as closing tag with above comment

**Review by StalkAltan — CHANGES_REQUESTED** [review]
Small request due to difference between GH and Nx.dev markdown rendering

**Comment by isaacplmann** [comment]
Ready for re-review

**Review by StalkAltan — APPROVED** [review]

**Pull request merged**